### ISSUE TITLE
Crash in Win10 TH2

### ISSUE THREAD

**Issue by FrankHB** [DynamoRIO/drmemory#1826]
Run: `drmemory -- cmd`

Output:
```
~~Dr.M~~ Dr. Memory version 1.9.0
~~Dr.M~~ Running "cmd.exe"
~~Dr.M~~ WARNING: application is missing line number information.
~~Dr.M~~ WARNING: application exited with abnormal code 0xc0000005
```

Log:
```
Dr. Memory version 1.9.0 build 4 built on Oct  2 2015 13:13:14
Dr. Memory results for pid 5636: "cmd.exe"
Application cmdline: "cmd.exe"
Recorded 116 suppression(s) from default D:\Tools\DrMemory\bin\suppress-default.txt

WARNING: application is missing line number information.
```

Run: `drmemory -debug -- cmd.exe`

Output:
```
~~Dr.M~~ Dr. Memory version 1.9.0
~~Dr.M~~ Running "cmd.exe"
~~Dr.M~~ ASSERT FAILURE (thread 15928): d:\drmemory_package\drsyscall\drsyscall_windows.c:341: false (syscall number table error detected)~~Dr.M~~ WARNING: application exited with abnormal code 0xffffffff
```

Log:
```
Dr. Memory version 1.9.0 build 4 built on Oct  2 2015 13:13:36
Dr. Memory results for pid 8684: "cmd.exe"
Application cmdline: "cmd.exe"
Recorded 116 suppression(s) from default D:\Tools\DrMemory\bin\suppress-default.txt
ASSERT FAILURE (thread 18740): d:\drmemory_package\drsyscall\drsyscall_windows.c:341: false (syscall number table error detected)
```

I'm using DrMemory release 1.9.0 in Windows 10 x64 build 10586.14. Things seem to be all the same for all programs.

**Comment by FrankHB** [comment]
Note there is a newline missing after the assertion message and before the warning in the console output.

**Comment by Unc3nZureD** [comment]
I am experiencing the exactly same problem.

**Comment by derekbruening** [comment]
So far I am unable to get my Win10 VM's to pull down the TH2 "November Update" via Windows Update despite them being non-upgrade RTM 10240 installs from months ago with Defer Upgrades un-selected.

**Comment by Unc3nZureD** [comment]
Well, I agree that it's update system is quite bad. Could I help somehow with my installation? Maybe try to enable developer updates.

**Comment by derekbruening** [comment]
It seems that TH2 is not available as an update for Windows 10 Enterprise and would have to be installed fresh.

**Comment by derekbruening** [comment]
> Could I help somehow with my installation?

With acquiring the system call lists, yes, but it would be best to test the changes locally.

**Comment by Unc3nZureD** [comment]
Can't you read it directly from the ntdll.dll if I upload it?

**Comment by derekbruening** [comment]
Yes, and we need to know the win32k.sys numbers as well, and all for X {x86,x64,wow64}, though as I said if they're changing the syscalls it would be best to run through our local tests for both x86 and x64 kernels to ensure nothing else changed that affects us.

As a first step, what does the global.<pid>.log say (only the results.txt was pasted above)?  There should be output from this line of code:

```
                WARN("WARNING: sysnum table "PIFX" != wrapper "PIFX" for %s\n",
                     e->num.number, num, name);
```

**Comment by Unc3nZureD** [comment]
I uploaded a log file which was produced when I tried to run a simple HxD hex editor with DrMemory. I couldn't find any lines which would say anything about the syscall. Are you sure it's about a changed syscall? 

[global.7536.txt](https://github.com/DynamoRIO/drmemory/files/56621/global.7536.txt)

**Comment by derekbruening** [comment]
From the initial entry above, this issue is about changed syscalls:
```
ASSERT FAILURE (thread 18740): d:\drmemory_package\drsyscall\drsyscall_windows.c:341: false (syscall number table error detected)
```

global.7536.txt is from release build -- please use debug build in general (`-debug`) for diagnosing problems.

**Comment by derekbruening** [comment]
Not only did they add new system calls without increasing the major version, minor version, or service pack version (explaining why the auto-handling in DR failed to even identify that this was an unknown OS version), but they changed the 64-bit wrappers.  For both reasons, fixing this will require DR changes too.

**Comment by mszmatuszewski** [comment]
Can we please get an update as to when a working version could be released?

**Comment by derekbruening** [comment]
Hopefully within a week or so.

**Comment by Unc3nZureD** [comment]
That would be awesome. I'm awaiting for the latest version :)

**Comment by derekbruening** [comment]
Primary fix is in 446007ec2c76fbab9220850b38aac48c538eee18 and initial tests work.

Please confirm that this build works for you on Windows 10 TH2 1511: https://build.chromium.org/p/client.drmemory/builds/DrMemory-Windows-1.9.16822-0x6d06080-sfx.exe

**Comment by Unc3nZureD** [comment]
According to a few quick test at me, it seems to be perfectly working :) Thank you!
